src/pattern_arx.cc

```diff
@@ -61,66 +63,70 @@ int contains_any(vector<wstring> tags, vector<wstring> candidates)
 	return 0;
 }
 
-void toLower(basic_string<wchar_t>& s)
+void toLower(UString& s)
 {
-   for (basic_string<wchar_t>::iterator p = s.begin(); p != s.end(); ++p)
-   {
-      *p = towlower(*p);
-   }
+	UString temp;
+	size_t i = 0;
+	size_t len = s.size();
+	UChar32 c;
+	while (i < len) {
+		U16_NEXT(s, i, len, c);
+		temp += u_tolower(c);
```

Comment:
Nope, sorry, that's invalid. Correct Unicode case folding requires looking at surrounding code points, so this needs the full `u_strToLower()` https://unicode-org.github.io/icu-docs/apidoc/dev/icu4c/ustring_8h.html#a1b516af424a82c73421754e63306b4b6